File: client/client-multi/client-objective-c/Kaa/channel/TransportProtocolId.m
/*
 * Copyright 2014-2016 CyberVision, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import "TransportProtocolId.h"

@implementation TransportProtocolId

- (instancetype)initWithId:(int32_t)id version:(int32_t)version {
    self = [super init];
    if (self) {
        _protocolId = id;
        _protocolVersion = version;
    }
    return self;
}

- (id)copyWithZone:(NSZone *)zone {
    return [[[self class] allocWithZone:zone] initWithId:self.protocolId version:self.protocolVersion];
}

- (NSUInteger)hash {
    NSUInteger prime = 31;
    NSUInteger result = 1;
    result = prime * result + self.protocolId;
    result = prime * result + self.protocolVersion;
    return result;
}

- (BOOL)isEqual:(id)object {
    
    if ([object isKindOfClass:[TransportProtocolId class]]) {
        TransportProtocolId *other = (TransportProtocolId*)object;
        if (other.protocolId == self.protocolId && other.protocolVersion == self.protocolVersion) {
            return YES;
        }
    }
    
    return NO;
}

- (NSString*)description {
    return [NSString stringWithFormat:@"TransportProtocolId [id:%i, version:%i]", self.protocolId, self.protocolVersion];
}

@end

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/TransportProtocolId.java
/*
 * Copyright 2014-2016 CyberVision, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.kaaproject.kaa.client.channel;

/**
 * Immutable class to represent transport ID. Holds references to transport
 * protocol id and transport protocol version
 *
 * @author Andrew Shvayka
 */
public final class TransportProtocolId {

  private final int id;
  private final int version;

  /**
   * All-args constructor.
   */
  public TransportProtocolId(int protocolId, int protocolVersion) {
    super();
    this.id = protocolId;
    this.version = protocolVersion;
  }

  public int getProtocolId() {
    return id;
  }

  public int getProtocolVersion() {
    return version;
  }

  @Override
  public int hashCode() {
    final int prime = 31;
    int result = 1;
    result = prime * result + id;
    result = prime * result + version;
    return result;
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) {
      return true;
    }
    if (obj == null) {
      return false;
    }
    if (getClass() != obj.getClass()) {
      return false;
    }
    TransportProtocolId other = (TransportProtocolId) obj;
    if (id != other.id) {
      return false;
    }
    if (version != other.version) {
      return false;
    }
    return true;
  }

  @Override
  public String toString() {
    return "TransportProtocolId [id=" + id + ", version=" + version + "]";
  }
}
